src/incubator/TextField/index.tsx

```diff
@@ -129,11 +130,12 @@ const TextField = (
               retainSpace
             />
           )}
-          <CharCounter
-            showCharCounter={showCharCounter}
-            maxLength={props.maxLength}
-            charCounterStyle={charCounterStyle}
-          />
+          {showCharCounter && !_.isUndefined(props.maxLength) && 
```

Comment:
I meant 'maxLength' should be required in CharCounter component so you don't need to check it here, the chack is in the component. 'showCharCounter' is of the TextField: 
{showCharCounter && 
  <CharCounter
    maxLength={props.maxLength}
    charCounterStyle={charCounterStyle}
 />}